src/NUnitEngine/Addins/nunit.v2.driver/XmlExtensions.cs

```diff
@@ -96,8 +96,8 @@ private static XmlNode AddTest(this XmlNode parent, ITest test, bool recursive)
             if (test.IsSuite)
                 thisNode.AddAttribute("type", test.TestType);
             thisNode.AddAttribute("id", test.TestName.TestID.ToString());
-            thisNode.AddAttribute("name", Escape(test.TestName.Name));
```

Comment:
I just added a testcase for this and am not seeing any differences in the output for either the Nunit 3 or Nunit2 driver. Both appear to be valid xml. 

The fixture I added the test to would've threw an exception if it was unable to read back the xml. 

```xml
<test-suite type="ParameterizedMethod" id="1057" name="MustBeEscaped" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped" runstate="Runnable" testcasecount="5" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.001000" total="5" passed="5" failed="0" inconclusive="0" skipped="0" asserts="0">
            <test-case id="1052" name="MustBeEscaped(&quot;\&quot;double quote\&quot;&quot;)" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped(&quot;\&quot;double quote\&quot;&quot;)" methodname="MustBeEscaped" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1219663533" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1053" name="MustBeEscaped(&quot;&lt; left bracket&quot;)" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped(&quot;&lt; left bracket&quot;)" methodname="MustBeEscaped" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1868752976" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1054" name="MustBeEscaped(&quot;\'single quote\'&quot;)" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped(&quot;\'single quote\'&quot;)" methodname="MustBeEscaped" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="262958866" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1055" name="MustBeEscaped(&quot;&gt; right bracket&quot;)" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped(&quot;&gt; right bracket&quot;)" methodname="MustBeEscaped" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1595902790" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1056" name="MustBeEscaped(&quot;&amp;amp&quot;)" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped(&quot;&amp;amp&quot;)" methodname="MustBeEscaped" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1912490567" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
          </test-suite>
          <test-suite type="ParameterizedMethod" id="1063" name="MustBeEscaped_CustomName" fullname="NUnit.Tests.TestNameEscaping.MustBeEscaped_CustomName" runstate="Runnable" testcasecount="5" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.001000" total="5" passed="5" failed="0" inconclusive="0" skipped="0" asserts="0">
            <test-case id="1058" name="&quot;" fullname="NUnit.Tests.TestNameEscaping.&quot;" methodname="MustBeEscaped_CustomName" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="670081083" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1059" name="&lt;" fullname="NUnit.Tests.TestNameEscaping.&lt;" methodname="MustBeEscaped_CustomName" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1143024897" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1060" name="&amp;" fullname="NUnit.Tests.TestNameEscaping.&amp;" methodname="MustBeEscaped_CustomName" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1109780594" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1061" name="&gt;" fullname="NUnit.Tests.TestNameEscaping.&gt;" methodname="MustBeEscaped_CustomName" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1435497444" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
            <test-case id="1062" name="'" fullname="NUnit.Tests.TestNameEscaping.'" methodname="MustBeEscaped_CustomName" classname="NUnit.Tests.TestNameEscaping" runstate="Runnable" seed="1460000646" result="Passed" start-time="2015-04-09 15:09:28Z" end-time="2015-04-09 15:09:28Z" duration="0.000000" asserts="0" />
          </test-suite>
        </test-suite>


```